### PR TITLE
Reinforce the WatchTower implementation

### LINKED ISSUE #321: Reinforce the WatchTower implementation
## Description 📹

**Replace println & eprintln by WatchTower (our own logger) as done in src/endpoints/quest_boost/get_pending_claims.rs**

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:
   ```bash
   git checkout -b fix-[issue-number]
   ```

2. **Implement Changes**:  
- [ ] Use the global search tool provided by your IDE to find all occurences of println and replace them, if possible, by our logger

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:
   ```bash
   git commit -m "Fix: [Short description of the fix]"
   ```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing `Close #[issue_id]`.
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by Michaelkingsdev**
Closes #321 

The following files were modified
1. utils.rs
2. rewards.rs
3. get_alt_protocol_stats.rs

Is there any need to modify `logger.rs`

### CODE REVIEW

**Comment by Michaelkingsdev** [comment]
@Marchand-Nicolas 
Review

File: src/utils.rs [hunk 1]
```diff
@@ -932,15 +932,16 @@ pub async fn check_if_unclaimed(
     calldata: Vec<FieldElement>,
     source: RewardSource,
 ) -> bool {
+    let logger = &state.logger;
     match read_contract(state, contract, selector, calldata).await {
         Ok(result) => result.get(0) == Some(&FieldElement::ZERO),
         Err(err) => {
-            eprintln!(
+            logger.info(format!(
```
Comment (Marchand-Nicolas) [inline src/utils.rs]: Please use logger.severe, because it's an error

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]
Looks very good! Please pull the changes made on the testnet branch, because they contain a fix for the build error you might have experienced. It will allow me to test it

File: src/utils.rs (hunk 1 above)
Comment (Michaelkingsdev) [reply src/utils.rs]: Alright

**Comment by Michaelkingsdev** [comment]
> Looks very good! Please pull the changes made on the testnet branch, because they contain a fix for the build error you might have experienced. It will allow me to test it

Done @Marchand-Nicolas

**Review by Marchand-Nicolas — APPROVED** [review]
Lgtm!

**Pull request merged**